### PR TITLE
Issue #4356: JavadocStyle - Check one line empty before tags in javadoc

### LINKED ISSUE #4356: Check one line empty before tags in javadoc
Oracle (in JDK they use it) suggests to insert one empty line in javadoc before tags (param, return etc) see http://www.oracle.com/technetwork/articles/java/index-137868.html. For example like this,

```
/**
 * Method for doing it.
 *
 * @param foo somefoo.
 */
public void doIt(Foo foo){}
```
So I suggest to add to checkstyle such rule.

### PR DESCRIPTION
**Pull request by gaurabdg**
Resolves #4356 

Diff report - https://gaurabdg.github.io/checkstyle-tester-reports/regression/JavadocStyle-lineBeforeTag/diff/

### CODE REVIEW

**Comment by gaurabdg** [comment]
@rnveach @romani Please review

**Comment by rnveach** [comment]
@gaurabdg Why is the functionality in JavadocStyle? I see nothing writtin in issue about this check. Where in our documentation does it say it should be doing this type of validation?

**Comment by gaurabdg** [comment]
@rnveach JavadocStyle seemed to be the best module to have this feature since it deals with the style of the javadoc comment. Should it be implemented as a separate check as asked by @Vladlis in the issue comment?

**Comment by gaurabdg** [comment]
> What is worse is JavadocStyle is using old regular expression parsing which is deprecated and we are trying to remove it.

@rnveach I used regex matching for this functionality since I integrated in JavadocStyle. Then shall I create a separate check using Javadoc AST?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java [hunk 1]
```diff
@@ -269,18 +278,21 @@
...
+    /** Message property key for the Empty Javadoc message. */
     public static final String MSG_EMPTY = "javadoc.empty";
 
-    /** Message property key for the Unclosed HTML message. */
+    /** Message property key for the No Javadoc end of Sentence Period message. */
     public static final String MSG_NO_PERIOD = "javadoc.noPeriod";
 
-    /** Message property key for the Unclosed HTML message. */
+    /** Message property key for the Incomplete Tag message. */
     public static final String MSG_INCOMPLETE_TAG = "javadoc.incompleteTag";
```
Comment (baratali) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java:291]: Can you please extract these changes to a new pull request?
This is not really related to the issue, so we can merge it faster in another PR.

**Comment by baratali** [comment]
The same feature is being implemented in https://github.com/checkstyle/checkstyle/pull/7932.
Looks like we will need to close this pull request.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java (hunk 1 above)
Comment (gaurabdg) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java:291]: Sure!

**Comment by baratali** [comment]
@gaurabdg , I have merged your another PR: https://github.com/checkstyle/checkstyle/pull/8139.
I'm going to close this PR, because https://github.com/checkstyle/checkstyle/pull/7932 implements same feature and uses 'new' Javadoc AST instead of regexp.
Thanks for your work. This PR will be counted as your completed GSoC task anyway ;)

**Comment by gaurabdg** [comment]
@baratali thanks!
